packages/api/src/main/java/com/incubator/edupayroll/service/teacher/TeacherService.java

```diff
@@ -30,4 +31,28 @@ public List<TeacherEntity> getAll(UserEntity user, int limit, int offset) {
 
         return page.get().toList();
     }
+
+    public TeacherEntity getById(UserEntity user, UUID uuid) {
+        return teacherRepository.getReferenceById(uuid);
```

Comment:
burada öncelikle `findById` metodunu kullanmalıyız.
mevcut teacher'i çektikten sonra null check yapmalıyız. eğer yoksa user not found fırlatmalıyız. 
varsa teacher bu kullanıcıya mı ait kontrolü yapmalıyız. eğer değilse access denied exception fırlatmalıyız. `utils/exception` altında bulunuyor.

ardından teacher'ı dönmeliyiz.